apps/socketoptions.cpp

```diff
@@ -26,55 +26,47 @@ SocketOption::Mode SrtConfigurePre(SRTSOCKET socket, string host, map<string, st
     vector<string> dummy;
     vector<string>& fails = failures ? *failures : dummy;
 
-    if ( options.count("passphrase") )
+    SocketOption::Mode mode = SocketOption::CALLER;
+    if (host != "--") // for this value it will always return CALLER
```

Comment:
`host != "--"` not clear when can this happen.

Reply:
I need this function to serve all purposes as before, but the handling of the "host" thing in srt-tunnel application isn't exactly what I need in this case. In my case I don't have any host to pass there, and whatever I pass would be misleading; simultaneously if I pass empty host, this function will configure options as for listener. I needed a mode in which I can call this function still the old way, without passing the host, but enforcing it will report caller mode always.

Reply:
Ok, added mode just before calling this function. Seems to work.